x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearningUsageTransportAction.java

```diff
@@ -112,26 +115,19 @@ protected void masterOperation(Task task, XPackUsageRequest request, ClusterStat
             listener::onFailure
         );
 
-        // Step 4. Extract usage from ingest statistics and gather trained model config count
+        // Step 5. Extract usage from ingest statistics and gather trained model config count
         ActionListener<NodesStatsResponse> nodesStatsListener = ActionListener.wrap(
             response -> {
                 addInferenceIngestUsage(response, inferenceUsage);
-                SearchRequestBuilder requestBuilder = client.prepareSearch(InferenceIndexConstants.INDEX_PATTERN)
-                    .setSize(0)
-                    .setQuery(QueryBuilders.boolQuery()
-                        .filter(QueryBuilders.termQuery(InferenceIndexConstants.DOC_TYPE.getPreferredName(), TrainedModelConfig.NAME)))
-                    .setTrackTotalHits(true);
-                ClientHelper.executeAsyncWithOrigin(client.threadPool().getThreadContext(),
-                    ClientHelper.ML_ORIGIN,
-                    requestBuilder.request(),
-                    trainedModelConfigCountListener,
-                    client::search);
+                GetTrainedModelsAction.Request getModelsRequest = new GetTrainedModelsAction.Request("*", Collections.emptyList(),
+                    Collections.emptySet());
+                client.execute(GetTrainedModelsAction.INSTANCE, getModelsRequest, trainedModelsListener);
```

Comment:
I think you want to set the page params here as well. I forget what the maximum size is off hand.

Reply:
Excellent point! Should be 10K.